pkgs/development/libraries/jabcode/default.nix

```diff
@@ -0,0 +1,46 @@
+{ stdenv
+, fetchFromGitHub
+, lib
+, subproject ? "library" # one of "library", "reader" or  "writer"
+, zlib, libpng, libtiff
+, jabcode 
+}:
+let
+  subdir = lib.getAttr subproject {
+    "library" = "jabcode";
+    "reader" = "jabcodeReader";
+    "writer" = "jabcodeWriter";
+  };
+in stdenv.mkDerivation rec {
+  pname = subproject;
+  version = "git-2020-05-13";
+  src = fetchFromGitHub {
+    repo = "jabcode";
+    owner = "jabcode";
+    rev = "a7c25d4f248078f257b014e31c791bfcfcd083e1";
+    sha256 = "1c4cv9b0d7r4bxzkwzdv9h651ziq822iya6fbyizm57n1nzdkk4s";
+  };
+
+  nativeBuildInputs =
+    [ zlib libpng libtiff ]
+    ++ lib.optionals (subproject != "library") [ jabcode ];
+
+  preConfigure = "cd src/${subdir}";
+
+  installPhase = if subproject == "library" then ''
```

Comment:
It seems like the library really should be its own derivation, it shares little code with the reader/writer.

Reply:
Not sure about that, it does share the source as well as some of the build inputs. Also since its all in the same repo upstream, if they decide to e.g. make changes to the build system at some point, they could become more coupled in the future.

Reply:
If you feel strongly about this, I can decouple them

Reply:
No, I don't feel very strongly. I think dependencies are likely to diverge, but your points are valid as well.